Weekly_Work/Week1/Code_changes-git/Metrics_Root_class.py

```diff
@@ -0,0 +1,202 @@
+                    clean_pr_list.append(clean_line)
+                        
+                        
+                self.clean_commit_df = pd.DataFrame(clean_commit_list)
+                self.clean_issue_df = pd.DataFrame(clean_issue_list)
+                self.clean_pr_df = pd.DataFrame(clean_pr_list)
+                
+            self.clean_dict = {
+                'commit': self.clean_commit_df,
+                'issue': self.clean_issue_df,
+                'pull_request': self.clean_pr_df
+            }
+       
+        if self.since:
+            for df in self.clean_dict.values():
+                df = df[df['created_date'] >= self._str_to_dt_other(self.since)]
+        else: 
+            self.since = self._get_date("since")
+            
+        if self.until:
+            for df in self.clean_dict.values():
+                df = df[df['created_date'] < self._str_to_dt_other(self.until)]
+        else: 
+            self.until = self._get_date("until")
+        
+    def _clean_commit(self, line):
+        cdata = line['data']
+        cleaned_line =  \
+        {
+            'repo': line['origin'],
+            'hash': cdata['commit'],
+            'author': cdata['Author'],
+            'category': "commit",
+            'created_date': self._str_to_dt_data(cdata['AuthorDate']),
+            'commit': cdata['Commit'],
+            'commit_date': self._str_to_dt_data(cdata['CommitDate']),
+            'files_no': len(cdata['files']),
+            'refs': cdata['refs'],
+            'parents': cdata['parents'],
+            'files': cdata['files']
+        }
+
+        actions = 0
+        for file in cdata['files']:
+            if 'action' in file:
+                actions += 1
+        cleaned_line['files_action'] = actions
+        if 'Merge' in cdata:
+            cleaned_line['merge'] = True
+        else:
+            cleaned_line['merge'] = False
+
+        return cleaned_line
+            
+    def _clean_issue(self, line):
+        repo_name = line['origin']
+        line_data = line['data']
+        cleaned_line =  \
+        {
+            'repo': repo_name,
+            'hash': line_data['id'],
+            'category': "issue",
+            'author': line_data['user']['login'],
+            'created_date': self._str_to_dt_data(line_data['created_at']),
+            'current_status': line_data['state']   
+        }
+        return cleaned_line
+
+            
+    def _clean_pr(self, line):
+        repo_name = line['origin']
+        line_data = line['data']
+        cleaned_line =  \
+        {
+            'repo': repo_name,
+            'hash': line_data['id'],
+            'category': "pull_request",
+            'author': line_data['user']['login'],
+            'created_date': self._str_to_dt_data(line_data['created_at']),
+            'current_status': line_data['state']   
+        }
+        return cleaned_line
+
+    def _is_source_code(self, commit):
+        """
```

Comment:
This is the kind of head comments that I mentioned earlier, and that it would be great having everywhere ;-)

Reply:
Great! Will do!